restler/engine/core/async_request_utilities.py

```diff
@@ -228,6 +244,8 @@ def try_async_poll(request_data, response, max_async_wait_time):
                 if get_response:
                     # Use response from the GET request for parsing. If the GET request failed,
                     # the caller will know to try and use the original PUT response for parsing
-                    response_to_parse = get_response
+                    response_to_parse.append(get_response)
```

Comment:
shouldn't this be 'responseS_to_parse.append(get_response) ?

Reply:
yes, this is a bug (in an error leg not exercised in manual testing -- JSONDecodeError) -- please fix this typo before merging.

Reply:
please fix the above